nats.go

```diff
@@ -496,9 +496,6 @@ type Conn struct {
 	respMux   *Subscription        // A single response subscription
 	respMap   map[string]chan *Msg // Request map for the response msg channels
 	respRand  *rand.Rand           // Used for generating suffix
-
-	// JetStream Contexts last account check.
-	jsLastCheck time.Time
```

Comment:
I think we all agreed that JS contexts should be lightweight and easy to create. In many tests I create multiple ones with different characteristics.

Question though, do we expect code to look like this all the time or folks will skip the AccountInfo and just check errors on the other API calls?

```go
nc, _ := nc.Connect("demo.nats.io")
js, _ := nc.JetStream()
if _, err := js.AccountInfo(); err != nil {}
```

Reply:
Help me understand: can the enabled/disabled nature of JetStream for this connection change overtime? Meaning, after calling JetStream() (and lets say we still do an account check inside, and it returned ok), is there a possibility that JS may no longer be enabled for this account? What if the connection reconnects to a server where this account does not have JS enabled, which likely would be misconfiguration?

If it can't change, then instead of a time check, it should be simply a "already checked and here is the result" type of things (say 2 booleans).

If it can change overtime and since all APIs will return "not enabled" if JS is or no longer is enabled, not sure what we gain by doing it inside JetStream(). Say you already have the context, but then JS is disabled, then the context is "valid" and yet all other APIs call will fail with "not enabled". Same if you recently got a context and we use time-check, then if we are within the window, JetStream() would return a context (which won't work when calling APIs), but if you happen to ask after the time-check has passed, this call would return "not enabled".

I am not against doing the account check inside JetStream() (with the time-check bug fixed), since it can give a "fail fast" behavior where if you get "not enabled", there is no need to proceed further, but it also can be non deterministic (as described above), and all APIs should be checked for success/failure anyway.

Reply:
due to reload or jwt update Js can be legitimately enabled disabled on the fly.
We have code that turns a no responder into the same error, so flapping because of that can happen as well.
If you want to fail fast, perhaps for debugging, I'd suggest call js.AccountInfo. 
I'd also suggest calling that function on retries. you fails for some reason now you can better determine if retrying makes sense or not.

Reply:
I think for users that want to ensure that JS is ready and available, a bit more code would be needed which might be better to have outside of the creation of the `JetStream()` context:

```go
nc, _ := nc.Connect("demo.nats.io")
js, _ := nc.JetStream()

ctx, done := context.WithTimeout(context.Background(), 5*time.Second)
defer done()

// IsJSAvailable()
for {
        select {
        case <-ctx.Done():
                return
        default:
        }

	if _, err := js.AccountInfo(); err != nil {
		// Handle temporary errors:
		//
		// - Quorum not ready
		// - JS API not yet ready
		// - No responders (temporary while JS booting up)

		// Then: timeout when too many failures or no response
	}
}
```

There would be some retries when this happens for example and backoff until is ready.